docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,625 @@
+
+This field of the root `Query` type must return a non-nullable [`_Service` type](#type-_service).
+
+For details, see [Enhanced introspection with `Query._service`](#enhanced-introspection-with-query_service).
+
+### `Query._entities`
+
+The graph router uses this root-level `Query` field to directly fetch fields of entities defined by a subgraph.
+
+This field must take a `representations` argument of type `[_Any!]!` (a non-nullable list of non-nullable [`_Any` scalars](#scalar-_any)). Its return type must be `[_Entity]!` (a non-nullable list of _nullable_ objects that belong to the [`_Entity` union](#union-_entity)).
+
+Each entry in the `representations` list  must be validated with the following rules:
+
+* A representation must include a `__typename` string field.
+* A representation must contain all fields included in the fieldset of a `@key` directive applied to the corresponding entity definition.
+
+For details, see [Resolving entity fields with `Query._entities`](#resolving-entity-fields-with-query_entities).
+
+### == Added types ==
+
+### `type _Service`
+
+This object type must have an `sdl: String!` field, which returns the SDL of the subgraph schema as a string.
+
+The returned schema string _must_ include all uses of federation-specific directives (`@key`, `@requires`, etc.), and it _must not_ include any definitions from the [subgraph schema specification](#subgraph-schema-specification).
+
+For details, see [Enhanced introspection with `Query._service`](#enhanced-introspection-with-query_service).
+
+### `union _Entity`
+
+**⚠️ This union type is generated dynamically based on the input subgraph schema!**
+
+This union's possible types must include all entities that the subgraph defines. It's the return type of the `Query._entities` field, which the graph router uses to directly access a subgraph's entity fields.
+
+For details, see [Defining the `_Entity` union](#defining-the-_entity-union).
+
+### `scalar _Any`
+
+This scalar is the type used for entity **representations** that the graph router passes to the `Query._entities` field. An `_Any` scalar is validated by matching its `__typename` and `@key` fields against entities defined in the subgraph schema.
+
+### `scalar FieldSet`
+
+This string-serialized scalar represents a set of fields that's passed to a federated directive, such as `@requires` or `@provides`.
+
+Grammatically, a `FieldSet` is a [selection set](http://spec.graphql.org/draft/#sec-Selection-Sets) minus the outermost curly braces. It can represent a single field (`"upc"`), multiple fields (`"id countryCode"`), and even nested selection sets (`"id organization { id }"`).
+
+### == Added directives ==
+
+### `@external`
+
+```graphql
+directive @external on FIELD_DEFINITION
+```
+
+The `@external` directive indicates that a particular subgraph _cannot_ resolve a particular entity field. This enables a subgraph to reference a _different_ subgraph's field _without_ also resolving that field.
```

Comment:
this is not completely true

```
type Foo @key("id") {
  id: ID!
  localBar: Bar @provides("baz") # resolves field locally even though it is marked as external
  remoteBar: Bar # resolves field remotely
}

extend type Bar @key("id) {
  id: ID!
  baz: String @external
}
```

Reply:
in fed v2 `@external` is only required for `@required` fields and in some cases (see above) `@provides` fields

Reply:
Right, I use better language for this in the other directives article and I'll make sure to review. The point being at least that a subgraph _usually_ can't resolve an `@external` field unless there's a path that allows it via `@provides`